File: src/generateurLabyrinthe/GenerateurLabyrinthe.scala
package generateurLabyrinthe

import scala.util._
import java.io.BufferedWriter
import java.io.BufferedReader
import java.io.Reader
import java.io.Writer
import java.io.File
import java.io.FileReader
import java.io.FileWriter
import graph._

class GenerateurLabyrinthe {

    def sauvegarderLabyrinthe(graph: Graph[Int]) = {
        val writer = new BufferedWriter(new FileWriter("ressources/" + graph.nom + ".dot"))
        writer.write(graph.adjacence.size + "\n")
        writer.write("graph " + graph.nom + " {\n")
        graph.adjacence.keys.foreach { i =>
            for(j <- 0 until graph.adjacence(i).size) {
                writer.write(i + " -> " + graph.adjacence(i)(j) + "\n")
            }
        }
        writer.write("}")
        writer.close()
    }
    def chargerLabyrinthe(cheminFichier: String): Graph[Int] = {
        val reader = new BufferedReader(new FileReader(cheminFichier))
        val nbNodes = reader.readLine.toInt
        println("nb Node : " + nbNodes)
        val premiereLigne = reader.readLine.toString
        val result = premiereLigne.split("\\s");
        val nomGraph = result(1)

        val graph = new Graph[Int](nomGraph, nbNodes)
        for(i <- 0 until nbNodes){
            val Array(key1, key2) = for(i <- reader.readLine split " -> ") yield i.toInt
            if(!graph.nodePresent(key1)) graph.addNode(key1, key1)
            if(!graph.nodePresent(key2)) graph.addNode(key2, key2)
            graph.addEdge(key1, key2, 1)
            graph.addEdge(key2, key1, 1)
        }
        graph
    }
}

object ClassMain {
    def main(args: Array[String]): Unit = {
        var generateur = new GenerateurLabyrinthe
        var graph = new Graph[Int]("graph1", 4)
        graph.addNode(0, 0)
        graph.addNode(1, 1)
        graph.addNode(2, 2)
        graph.addNode(3, 3)

        graph.addEdge(0, 1, 1)
        graph.addEdge(1, 2, 1)
        graph.addEdge(2, 0, 1)
        graph.addEdge(2, 3, 1)

        generateur.sauvegarderLabyrinthe(graph)
        val graphCharge = generateur.chargerLabyrinthe("ressources/graph1.dot")
        graphCharge.display
    }
}

File: developpement/src/main/java/javabyrinthe/labyrinthe/GenerateurLabyrinthe.java
package javabyrinthe.labyrinthe;

import graph.Graph;

import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

import javabyrinthe.tools.Coordonnee;

public class GenerateurLabyrinthe {
	
	public Labyrinthe chargerGraphe(String cheminFichier) throws FileNotFoundException {
		try {
			BufferedReader reader = new BufferedReader(new FileReader(cheminFichier));
			Integer nbEdges = Integer.parseInt(reader.readLine());
			System.out.println("nb edges : " + nbEdges);
			String premiereLigne = reader.readLine().toString();
			String[] result = premiereLigne.split(" ");
			String nomGraph = result[1];
			
			Graph graph = new Graph(nomGraph, nbEdges);
			for(int i = 0; i < nbEdges; i++) {
				String[] keys = reader.readLine().split(" -> ");
				if(!graph.nodePresent(Integer.parseInt(keys[0])))
					graph.addNode(Integer.parseInt((keys[0])));
				graph.addEdge(Integer.parseInt(keys[0]), Integer.parseInt(keys[1]), 1);
			}
			reader.close();
			return new Labyrinthe(graph, new Coordonnee(0, 0), new Coordonnee(graph.getSize() / 2, graph.getSize() / 2));
		} catch (IOException e) {
			e.printStackTrace();
			return null;
		}
	}
}
